src/nuclide.cpp

```diff
@@ -404,7 +497,245 @@ double Nuclide::elastic_xs_0K(double E) const
+      + f*xs(i_grid + 1,XS_ABSORPTION);
+
+    if (fissionable_) {
+      // Calculate microscopic nuclide total cross section
+      micro_xs.fission = (1.0 - f)*xs(i_grid,XS_FISSION)
+            + f*xs(i_grid + 1,XS_FISSION);
+
+      // Calculate microscopic nuclide nu-fission cross section
+      micro_xs.nu_fission = (1.0 - f)*xs(i_grid,XS_NU_FISSION)
+        + f*xs(i_grid + 1,XS_NU_FISSION);
+    } else {
+      micro_xs.fission = 0.0;
+      micro_xs.nu_fission = 0.0;
+    }
+
+    // Calculate microscopic nuclide photon production cross section
+    micro_xs.photon_prod = (1.0 - f)*xs(i_grid,XS_PHOTON_PROD)
+      + f*xs(i_grid + 1,XS_PHOTON_PROD);
+
+    // Depletion-related reactions
+    if (simulation::need_depletion_rx) {
+      // Initialize all reaction cross sections to zero
+      for (double& xs_i : micro_xs.reaction) {
+        xs_i = 0.0;
+      }
+
+      for (int j = 0; j < DEPLETION_RX.size(); ++j) {
+        // If reaction is present and energy is greater than threshold, set the
+        // reaction xs appropriately
+        int i_rx = reaction_index_[DEPLETION_RX[j]];
+        if (i_rx >= 0) {
+          const auto& rx = reactions_[i_rx];
+          const auto& rx_xs = rx->xs_[i_temp].value;
+
+          // Physics says that (n,gamma) is not a threshold reaction, so we don't
+          // need to specifically check its threshold index
+          if (j == 0) {
+            micro_xs.reaction[0] = (1.0 - f)*rx_xs[i_grid]
+              + f*rx_xs[i_grid + 1];
+            continue;
+          }
+
+          // TODO: off-by-one
+          int threshold = rx->xs_[i_temp].threshold - 1;
+          if (i_grid >= threshold) {
+            micro_xs.reaction[j] = (1.0 - f)*rx_xs[i_grid - threshold] +
+              f*rx_xs[i_grid - threshold + 1];
+          } else if (j >= 3) {
+            // One can show that the the threshold for (n,(x+1)n) is always
+            // higher than the threshold for (n,xn). Thus, if we are below
+            // the threshold for, e.g., (n,2n), there is no reason to check
+            // the threshold for (n,3n) and (n,4n).
+            break;
+          }
+        }
+      }
+    }
+  }
+
+  // Initialize sab treatment to false
+  micro_xs.index_sab = -1;
+  micro_xs.sab_frac = 0.0;
+
+  // Initialize URR probability table treatment to false
+  micro_xs.use_ptable = false;
+
+  // If there is S(a,b) data for this nuclide, we need to set the sab_scatter
+  // and sab_elastic cross sections and correct the total and elastic cross
+  // sections.
+
+  if (i_sab >= 0) this->calculate_sab_xs(i_sab, E, sqrtkT, sab_frac);
+
+  // If the particle is in the unresolved resonance range and there are
+  // probability tables, we need to determine cross sections from the table
+  if (settings::urr_ptables_on && urr_present_ && !use_mp) {
+    int n = urr_data_[i_temp].n_energy_;
+    if ((E > urr_data_[i_temp].energy_(0)) &&
+        (E < urr_data_[i_temp].energy_(n-1))) {
```

Comment:
Can we use `micro_xs.index_temp` here instead of `i_temp` to avoid that awkward declaration of `i_temp` outside the `if` block?